source/common/modules/markdown-editor/util/strip-duplicate-spaces.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        stripDuplicateSpaces function
+ * CVM-Role:        Utility function
+ * Maintainers:     Rich Douglas
+ * License:         GNU GPL v3
+ *
+ * Description:     This function strips duplicate spaces from a selected region
+ *                  of text.
+ *
+ * END HEADER
+ */
+
+import { type EditorView } from '@codemirror/view'
+import { type ChangeSpec } from '@codemirror/state'
+
+/**
+ * Utility to strip duplicate spaces from a selected region of text.
+ *
+ * @param {EditorView} view - The editor view
+ */
+export function stripDuplicateSpaces (view: EditorView): void {
+  const changes = view.state.selection.ranges
+    .reduce((allChanges, { from, to }) => {
+      const text = view.state.sliceDoc(from, to)
+
+      const strippedText = text.replaceAll(/ {2,}/g, ' ')
+
+      if (strippedText !== text) {
+        allChanges.push({ from, to, insert: strippedText })
+      }
+
+      return allChanges
+    }, [] as ChangeSpec[])
+
+  if (changes.length !== 0) {
+    view.dispatch({ changes })
+  }
+}
```

Comment:
My natural inclination is to write (unit) tests for code.

I can't find any existing (unit) tests for CodeMirror-related code and some (cursory) Googling doesn't yield anything for (unit) testing CodeMirror.

So, I haven't included any tests with this code.

I'll continue to read up on CodeMirror: perhaps there are some integrtion-level testing approaches that we (I) can add to build up a battery of tests around editor functionality?

> I did consider extracting the "replace duplicates" code into a pure function that's easy to test because it has no CodeMirror dependency… but mmm, I can't find any examples of such in the existing codebase so I didn't do that either.

Happy to take your steer here Hendrik 🙇‍♂️✌️